platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/Shadowgraph.java

```diff
@@ -122,6 +132,7 @@ public Shadowgraph(@NonNull final PlatformContext platformContext, @NonNull fina
 
         this.metrics = new ShadowgraphMetrics(platformContext);
         this.numberOfNodes = addressBook.getSize();
+        this.intakeEventCounter = Objects.requireNonNull(intakeEventCounter);
```

Comment:
should the intake counter be part of the shadowgraph? seems misplaced

Reply:
The goal of the intake event counter is to track the number of events we have received through gossip but that have not yet been inserted into the shadowgraph. The implementation in develop is decrementing the count in the consensus engine, which was not quite correct. Previously the consensus engine was "after" the shadowgraph, so this got the job done technically speaking. Now that the shadowgraph is not before the consensus engine, it's important to move the decrementing of the count to the proper location.

Reply:
Lazar and I discussed this, and he now agrees that this is an ok way to do it for now. Marking comment as resolved.